azkaban-exec-server/src/main/java/azkaban/execapp/FlowExecutionMonitorManager.java

```diff
@@ -0,0 +1,116 @@
+package azkaban.execapp;
+
+import azkaban.sla.SlaOption;
+import azkaban.trigger.Condition;
+import azkaban.trigger.ConditionChecker;
+import azkaban.trigger.TriggerAction;
+import azkaban.trigger.builtin.KillExecutionAction;
+import azkaban.trigger.builtin.SlaAlertAction;
+import azkaban.trigger.builtin.SlaChecker;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.PriorityBlockingQueue;
+import org.apache.log4j.Logger;
+
+
+public class FlowExecutionMonitorManager extends Thread {
+  private static Logger logger = Logger.getLogger(FlowExecutionMonitorManager.class);
+  private BlockingQueue<Trigger> triggers = null;
+  private static final Duration CHECK_INTERVAL = Duration.ofMillis(15 * 1000);
+
+  public FlowExecutionMonitorManager() {
+    this.setName("FlowExecutionMonitorManager-Thread");
+    this.setDaemon(true);
+    this.triggers = new PriorityBlockingQueue<>(1, new TriggerComparator());
+  }
+
+  public Trigger[] getAllTriggers() {
+    return triggers.toArray(new Trigger[triggers.size()]);
+  }
+
+  private class TriggerComparator implements Comparator<Trigger> {
+    @Override
+    public int compare(Trigger arg0, Trigger arg1) {
+      long first = arg1.getNextCheckTime();
+      long second = arg0.getNextCheckTime();
+
+      if (first == second) {
+        return 0;
+      } else if (first < second) {
+        return 1;
+      }
+      return -1;
+    }
+  }
+
+  public void addMonitor(int execId, List<SlaOption> slaOptions) {
```

Comment:
nit:
I have read this logic earlier in the code. It still goes right over my head. Some doc here briefly describing what it does would be helpful. Else breaking it down into smaller functions would make it easy to read.